### ISSUE TITLE
ClickHouse suddenly stopped accepting incoming client connections

### ISSUE THREAD

**Issue by MikageSawatari** [ClickHouse/ClickHouse#16691]
**Describe the bug**
ClickHouse suddenly stopped accepting incoming client connections. We
haven't managed to identify the cause as of now, and think it may be
an unknown bug. We don't have reproduction steps either, as the
problem has occured only once.

**ClickHouse Version**
20.3.15 revision 54433

**Configuration**
`<max_connections>4096</max_connections>`

**What happened**

ClickHouse stopped accepting client connections while still
running. The native client could not connect to the server, even
though there was a log entry in the server log that it accepted a
connection. An HTTP client timed out while attempting to post a
request.

`netstat` showed that the server was listening on both 9000 and
8123. The problem went away after restarting the server, and it hasn't
occured since then.

**Additional context**

Neither `metric_log` nor `text_log` had any entries at the time when
the problem happened. Here are logs before and after the incident.

1 minute before the incident:

* TCPConnection
    * Average: 1.288, Median: 1, Maximum: 2
* MySQLConnection
    * Avg: 0, Med: 0, Max: 0
* HTTPConnection
    * Avg: 3.941, Med: 1, Max: 22
* InterserverConnection
    * Avg: 0, Med: 0, Max: 0

1 minute after restarting the server:

* TCPConnection
    * Average: 0, Median: 0, Maximum: 0
* MySQLConnection
    * Avg: 0, Med: 0, Max: 0
* HTTPConnection
    * Avg: 2.067, Med: 1, Max: 12
* InterserverConnection
    * Avg: 0.016, Med: 0, Max: 1

When the server was running normally:

* TCPConnection
    * Average: 2, Median: 2, Maximum: 2
* MySQLConnection
    * Avg: 0, Med: 0, Max: 0
* HTTPConnection
    * Avg: 2.610, Med: 0, Max: 23
* InterserverConnection
    * Avg: 0, Med: 0, Max: 1

Logs while it was running normally:

```
2020-11-01 00:00:01     Trace   HTTPHandler-factory     HTTP Request for HTTPHandler-factory. Method: POST, Address: ************, User-Agent: hackney/1.15.2, Length: 171, Content Type: application/octet-stream, Transfer Encoding: identity
2020-11-01 00:00:01     Trace   HTTPHandler     Request URI: /
2020-11-01 00:00:01     Debug   executeQuery    (from *********, user: ******) ALTER TABLE xxx.xxxxxx DELETE WHERE xxxxxx = \'**********\' AND xxxxxx < (toUInt64(toDateTime(\'2019-11-01 00:00:00\')) * 1000000) FORMAT JSONCompact
2020-11-01 00:00:01     Trace   AccessRightsContext (******)  Access granted: DELETE ON xxx.xxxxxx
2020-11-01 00:00:01     Debug   InterpreterSelectQuery  MergeTreeWhereOptimizer: condition "isZeroOrNull((xxxxxx = \'************\') AND (xxxxxx < (toUInt64(toDateTime(\'2019-11-01 00:00:00\')) * 1000000)))" moved to PREWHERE
2020-11-01 00:00:01     Debug   sms.send_request (ReplicatedMergeTreeQueue)     Pulling 1 entries to queue: log-0000250427 - log-0000250427
2020-11-01 00:00:01     Trace   HTTPHandler-factory     HTTP Request for HTTPHandler-factory. Method: POST, Address: ***********, User-Agent: hackney/1.15.2, Length: 347, Content Type: application/octet-stream, Transfer Encoding: identity
2020-11-01 00:00:01     Trace   HTTPHandler     Request URI: /
2020-11-01 00:00:01     Debug   executeQuery    (from ***********, user: ******)                SELECT toUInt32(SUM(xxxxxx))               FROM xxx.xxxxxx            WHERE xxxxxx = \'**************\'AND toDate(xxxxxx/1000000) >= toDate(\'2020-10-01\')AND toDate(\'2020-11-01\') > toDate(xxxxxx/1000000)AND xxxxxx = 1  AND xxxxxx != \'****\' AND xxxxxx != \'****\' AND xxxxxx != \'****\' AND xxxxxx != \'****\'
2020-11-01 00:00:01     Debug   InterpreterSelectQuery  MergeTreeWhereOptimizer: condition "xxxxxx = 1" moved to PREWHERE
2020-11-01 00:00:01     Trace   AccessRightsContext (******)  Access granted: SELECT(xxxxxx, xxxxxx, xxxxxx, xxxxxx, xxxxxx) ON xxx.xxxxxx
```

Logs just before the incident:

```
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler-factory        HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: ***********, User-Agent: none, Content Type: , Transfer Encoding: identity
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler        Request URI: ?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Fdelivery_records2%2Freplicas%2Fclickhouse-1&part=202010_77098_77098_0&client_protocol_version=2&compress=false
2020-10-30 15:34:47     Trace   sms_backend.delivery_records2 (Replicated PartsService) Sending part 202010_77098_77098_0
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler-factory        HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: ***********, User-Agent: none, Content Type: , Transfer Encoding: identity
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler        Request URI: ?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Fevent_log2%2Freplicas%2Fclickhouse-1&part=202010_302590_302590_0&client_protocol_version=2&compress=false
2020-10-30 15:34:47     Trace   xxx.xxxxxx (Replicated PartsService)        Sending part 202010_302590_302590_0
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler-factory        HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: ***********, User-Agent: none, Content Type: , Transfer Encoding: identity
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler        Request URI: ?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Fclick_log%2Freplicas%2Fclickhouse-1&part=202010_133109_133109_0&client_protocol_version=2&compress=false
2020-10-30 15:34:47     Trace   xxx.xxxxxx (Replicated PartsService)       Sending part 202010_133109_133109_0
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler-factory        HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: ***********, User-Agent: none, Content Type: , Transfer Encoding: identity
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler        Request URI: ?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Fsend_request%2Freplicas%2Fclickhouse-1&part=202010_77352_77352_0&client_protocol_version=2&compress=false
2020-10-30 15:34:47     Trace   sms.send_request (Replicated PartsService)      Sending part 202010_77352_77352_0
2020-10-30 15:34:47     Debug   xxx.xxxxxx (MergerMutator)  Selected 6 parts from 202010_302045_302584_108 to 202010_302589_302589_0
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler-factory        HTTP Request for InterserverIOHTTPHandler-factory. Method: POST, Address: ***********, User-Agent: none, Content Type: , Transfer Encoding: identity
2020-10-30 15:34:47     Trace   InterserverIOHTTPHandler        Request URI: ?endpoint=DataPartsExchange%3A%2Fclickhouse%2Ftables%2Fsend_request_for_monitor%2Freplicas%2Fclickhouse-1&part=202010_77049_77049_0&client_protocol_version=2&compress=false
```

There were no entries like `AccessRightsContext (******) Access
granted:` upon receiving an HTTP request. The server completely
stopped emitting any logs within 1 second after that.

**Comment by amosbird** [comment]
I've observed a similar issue but only with HTTPConnection in version 20.8.1.1. ClickHouse suddenly can only have one active HTTPConnection and queues up any other requests. However this issue is very rare and I haven't seen it for more than one month now.

**Comment by MikageSawatari** [comment]
The server we reported before is actually replicated, and our
application accesses only one of the servers. The issue regarding
incoming connections also happened to the one which was usually not
accessed by the application.

The ChangeLog of ClickHouse mentions several deadlock issues regarding
`text_log`. We run ClickHouse servers with `text_log` enabled. Can it
possibly be the cause of this connection issue?

> Enabled system.text_log in stress test to find more bugs. #13855 (Nikita Mikhaylov).
> Fixed deadlock in system.text_log. #12452 (alexey-milovidov). It is a part of #12339. This fixes #12325. #13386 (Nikita Mikhaylov).
> Fixed the deadlock if text_log is enabled. #12452 (alexey-milovidov).

Here is a summary of the situation and logs when the problem happened:

Number of connections (from `system.metric_log`)

- 1 minute before the incident

    - TCPConnection

        - Average: 0, Median: 0, Maximum: 0

    - MySQLConnection

        - Avg: 0, Med: 0, Max: 0

    - HTTPConnection

        - Avg: 0, Med: 0, Max: 0

    - InterserverConnection

        - Avg: 0, Med: 0, Max: 0

- There were no logs at the time when it happened.

- 1 minute after the incident

    - TCPConnection

        - Average: 0, Median: 0, Maximum: 0

    - MySQLConnection

        - Avg: 0, Med: 0, Max: 0

    - HTTPConnection

        - Avg: 0, Med: 0, Max: 0

    - InterserverConnection

        - Avg: 0, Med: 0, Max: 0

- When it's running normally

    - TCPConnection

        - Average: 0, Median: 0, Maximum: 0

    - MySQLConnection

        - Avg: 0, Med: 0, Max: 0

    - HTTPConnection

        - Avg: 0, Med: 0, Max: 0

    - InterserverConnection

        - Avg: 0, Med: 0, Max: 0


`text_log` just before the problem happened (there were no logs in
`system.text_log` until we restarted the server):

```
2020-11-14 09:17:29     Reserving 1.00 MiB on disk `default`, having unreserved 150.77 GiB.
2020-11-14 09:17:29     Renaming temporary part tmp_insert_202011_622925_622925_0 to 202011_623118_623118_0.
2020-11-14 09:17:29     Flushing system log
2020-11-14 09:17:29     Reserving 1.00 MiB on disk `default`, having unreserved 150.77 GiB.
2020-11-14 09:17:29     Renaming temporary part tmp_insert_202011_649075_649075_0 to 202011_649219_649219_0.
2020-11-14 09:17:29     Selected 2 parts from 202011_644015_649214_5189 to 202011_649215_649215_0
2020-11-14 09:17:29     Reserving 1.30 MiB on disk `default`, having unreserved 150.77 GiB.
2020-11-14 09:17:29     Merging 2 parts: from 202011_644015_649214_5189 to 202011_649215_649215_0 into tmp_merge_202011_644015_649215_5190 with type Wide
2020-11-14 09:17:29     Selected MergeAlgorithm: Horizontal
2020-11-14 09:17:29     Reading 12 marks from part 202011_644015_649214_5189, total 10890 rows starting from the beginning of the part
2020-11-14 09:17:29     Reading 2 marks from part 202011_649215_649215_0, total 1 rows starting from the beginning of the part
2020-11-14 09:17:29     Merge sorted 10891 rows, containing 37 columns (37 merged, 0 gathered) in 0.05 sec., 231721.58 rows/sec., 341.78 MB/sec.
2020-11-14 09:17:29     Renaming temporary part tmp_merge_202011_644015_649215_5190 to 202011_644015_649215_5190.
2020-11-14 09:17:29     Merged 2 parts: from 202011_644015_649214_5189 to 202011_649215_649215_0
2020-11-14 09:17:31     Flushing system log
2020-11-14 09:17:31     Reserving 1.00 MiB on disk `default`, having unreserved 150.76 GiB.
2020-11-14 09:17:31     Renaming temporary part tmp_insert_202011_649080_649080_0 to 202011_649261_649261_0.
2020-11-14 09:17:31     Found 7 old parts to remove.
2020-11-14 09:17:31     Removing 7 old parts from ZooKeeper
2020-11-14 09:17:31     Removed 7 old parts from ZooKeeper. Removing them from filesystem.
2020-11-14 09:17:31     Selected 5 parts from 202011_57326_57334_2 to 202011_57338_57338_0
2020-11-14 09:17:31     Pulling 1 entries to queue: log-0000466364 - log-0000466364
2020-11-14 09:17:31     Pulled 1 entries to queue.
2020-11-14 09:17:31     Removed 7 old parts
2020-11-14 09:17:31     Executing log entry to merge parts 202011_57326_57334_2, 202011_57335_57335_0, 202011_57336_57336_0, 202011_57337_57337_0, 202011_57338_57338_0 to 202011_57326_57338_3
2020-11-14 09:17:31     Reserving 1.00 MiB on disk `default`, having unreserved 150.76 GiB.
2020-11-14 09:17:31     Merging 5 parts: from 202011_57326_57334_2 to 202011_57338_57338_0 into tmp_merge_202011_57326_57338_3 with type Wide
2020-11-14 09:17:31     Checking 104 blocks (4 are not cached) to clear old ones from ZooKeeper.
2020-11-14 09:17:31     Selected MergeAlgorithm: Horizontal
2020-11-14 09:17:31     Reading 2 marks from part 202011_57326_57334_2, total 9 rows starting from the beginning of the part
2020-11-14 09:17:31     Reading 2 marks from part 202011_57335_57335_0, total 1 rows starting from the beginning of the part
2020-11-14 09:17:31     Reading 2 marks from part 202011_57336_57336_0, total 1 rows starting from the beginning of the part
2020-11-14 09:17:31     Reading 2 marks from part 202011_57337_57337_0, total 1 rows starting from the beginning of the part
2020-11-14 09:17:31     Reading 2 marks from part 202011_57338_57338_0, total 1 rows starting from the beginning of the part
2020-11-14 09:17:31     Merge sorted 13 rows, containing 17 columns (17 merged, 0 gathered) in 0.02 sec., 569.59 rows/sec., 0.37 MB/sec.
2020-11-14 09:17:31     Renaming temporary part tmp_merge_202011_57326_57338_3 to 202011_57326_57338_3.
2020-11-14 09:17:31     Merged 5 parts: from 202011_57326_57334_2 to 202011_57338_57338_0
2020-11-14 09:17:31     The part /clickhouse/tables/send_result/replicas/clickhouse-1/parts/202011_57326_57338_3 on a replica suddenly appeared, will recheck checksums
```

Logs written to a log file at the time when the server stopped writing logs to `system.text_log`:

```
2020.11.14 09:17:29.809257 [ 21643 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_649215_649215_0, total 1 rows starting from the beginning of the part
2020.11.14 09:17:29.848864 [ 21643 ] {} <Debug> system.query_thread_log (MergerMutator): Merge sorted 10891 rows, containing 37 columns (37 merged, 0 gathered) in 0.05 sec., 231721.58 rows/sec., 341.78 MB/sec.
2020.11.14 09:17:29.852059 [ 21643 ] {} <Trace> system.query_thread_log: Renaming temporary part tmp_merge_202011_644015_649215_5190 to 202011_644015_649215_5190.
2020.11.14 09:17:29.852340 [ 21643 ] {} <Trace> system.query_thread_log (MergerMutator): Merged 2 parts: from 202011_644015_649214_5189 to 202011_649215_649215_0
2020.11.14 09:17:31.092420 [ 21646 ] {} <Trace> SystemLog (system.query_log): Flushing system log
2020.11.14 09:17:31.094941 [ 21646 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 150.76 GiB.
2020.11.14 09:17:31.108994 [ 21646 ] {} <Trace> system.query_log: Renaming temporary part tmp_insert_202011_649080_649080_0 to 202011_649261_649261_0.
2020.11.14 09:17:31.333804 [ 21657 ] {} <Trace> sms.web_log: Found 7 old parts to remove.
2020.11.14 09:17:31.333870 [ 21657 ] {} <Debug> sms.web_log: Removing 7 old parts from ZooKeeper
2020.11.14 09:17:31.338886 [ 21657 ] {} <Debug> sms.web_log: Removed 7 old parts from ZooKeeper. Removing them from filesystem.
2020.11.14 09:17:31.339684 [ 21664 ] {} <Debug> sms.send_result (MergerMutator): Selected 5 parts from 202011_57326_57334_2 to 202011_57338_57338_0
2020.11.14 09:17:31.345309 [ 21656 ] {} <Debug> sms.send_result (ReplicatedMergeTreeQueue): Pulling 1 entries to queue: log-0000466364 - log-0000466364
2020.11.14 09:17:31.347831 [ 21656 ] {} <Debug> sms.send_result (ReplicatedMergeTreeQueue): Pulled 1 entries to queue.
2020.11.14 09:17:31.350773 [ 21657 ] {} <Debug> sms.web_log: Removed 7 old parts
2020.11.14 09:17:31.351146 [ 21638 ] {} <Trace> sms.send_result: Executing log entry to merge parts 202011_57326_57334_2, 202011_57335_57335_0, 202011_57336_57336_0, 202011_57337_57337_0, 202011_57338_57338_0 to 202011_57326_57338_3
2020.11.14 09:17:31.351200 [ 21638 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 150.76 GiB.
2020.11.14 09:17:31.351229 [ 21638 ] {} <Debug> sms.send_result (MergerMutator): Merging 5 parts: from 202011_57326_57334_2 to 202011_57338_57338_0 into tmp_merge_202011_57326_57338_3 with type Wide
2020.11.14 09:17:31.354571 [ 21657 ] {} <Trace> sms.web_log (ReplicatedMergeTreeCleanupThread): Checking 104 blocks (4 are not cached) to clear old ones from ZooKeeper.
2020.11.14 09:17:31.360272 [ 21638 ] {} <Debug> sms.send_result (MergerMutator): Selected MergeAlgorithm: Horizontal
2020.11.14 09:17:31.360321 [ 21638 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_57326_57334_2, total 9 rows starting from the beginning of the part
2020.11.14 09:17:31.362237 [ 21638 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_57335_57335_0, total 1 rows starting from the beginning of the part
2020.11.14 09:17:31.364028 [ 21638 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_57336_57336_0, total 1 rows starting from the beginning of the part
2020.11.14 09:17:31.365812 [ 21638 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_57337_57337_0, total 1 rows starting from the beginning of the part
2020.11.14 09:17:31.367537 [ 21638 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 202011_57338_57338_0, total 1 rows starting from the beginning of the part
2020.11.14 09:17:31.374060 [ 21638 ] {} <Debug> sms.send_result (MergerMutator): Merge sorted 13 rows, containing 17 columns (17 merged, 0 gathered) in 0.02 sec., 569.59 rows/sec., 0.37 MB/sec.
2020.11.14 09:17:31.375389 [ 21638 ] {} <Trace> sms.send_result: Renaming temporary part tmp_merge_202011_57326_57338_3 to 202011_57326_57338_3.
2020.11.14 09:17:31.375512 [ 21638 ] {} <Trace> sms.send_result (MergerMutator): Merged 5 parts: from 202011_57326_57334_2 to 202011_57338_57338_0
2020.11.14 09:17:31.381921 [ 21638 ] {} <Information> sms.send_result: The part /clickhouse/tables/send_result/replicas/clickhouse-1/parts/202011_57326_57338_3 on a replica suddenly appeared, will recheck checksums
2020.11.14 09:17:31.654144 [ 21626 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 150.76 GiB.
2020.11.14 09:17:31.664895 [ 21626 ] {} <Trace> system.text_log: Renaming temporary part tmp_insert_202011_861798_861798_0 to 202011_998090_998090_0.
2020.11.14 09:17:32.279946 [ 21657 ] {} <Debug> sms.web_log (ReplicatedMergeTreeQueue): Pulling 1 entries to queue: log-0000930336 - log-0000930336
2020.11.14 09:17:32.279991 [ 21653 ] {} <Debug> sms.event_log2 (ReplicatedMergeTreeQueue): Pulling 1 entries to queue: log-0001304796 - log-0001304796
```

Logs written to a file, before and after restarting the server:

```
2020.11.14 09:17:41.436135 [ 21643 ] {} <Trace> system.query_thread_log: Found 4 old parts to remove.
2020.11.14 09:17:45.764296 [ 16993 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16671, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:18:04.779886 [ 5026 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16734, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:18:19.793658 [ 5053 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16770, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:18:34.813426 [ 4979 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16838, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:18:49.818631 [ 4981 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16877, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:19:04.823427 [ 21685 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16917, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:19:19.829355 [ 4962 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:16977, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:19:34.833328 [ 4978 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:17006, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:19:49.839244 [ 5134 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:17059, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:20:04.843679 [ 5334 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:17097, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
....
2020.11.14 09:35:20.149380 [ 2010 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:29281, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:35:35.153718 [ 2026 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:29543, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:35:46.423574 [ 2040 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 172.28.0.9:33224, User-Agent: check_http/v2.2.1 (nagios-plugins 2.2.1), Content Type: , Transfer Encoding: identity
2020.11.14 09:35:50.158607 [ 2044 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:29701, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:36:05.163070 [ 2069 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:29929, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:36:20.167603 [ 2083 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:30123, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:36:25.130320 [ 21625 ] {} <Information> Application: Received termination signal (Terminated)
2020.11.14 09:36:35.171574 [ 2186 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:30419, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:36:50.176352 [ 2206 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:30660, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:37:05.181241 [ 2233 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:30905, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:37:20.186130 [ 2248 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:31120, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:37:35.190329 [ 2267 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: GET, Address: 192.168.249.254:31314, User-Agent: check_http/v2.0.3 (nagios-plugins 2.0.3), Content Type: , Transfer Encoding: identity
2020.11.14 09:38:07.921875 [ 2343 ] {} <Trace> Pipe: Pipe capacity is 1.00 MiB
2020.11.14 09:38:07.925503 [ 2343 ] {} <Information> : Starting ClickHouse 20.3.15.133 with revision 54433
2020.11.14 09:38:07.925622 [ 2343 ] {} <Information> Application: starting up
2020.11.14 09:38:07.937032 [ 2343 ] {} <Information> StatusFile: Status file /usr/local/ymir/clickhousevar/data/status already exists - unclean restart. Contents:
PID: 21621
Started at: 2020-08-31 11:18:27
Revision: 54433
```

```
2020.11.14 09:38:07.937247 [ 2343 ] {} <Debug> Application: rlimit on number of file descriptors is 262144
2020.11.14 09:38:07.937266 [ 2343 ] {} <Debug> Application: Initializing DateLUT.
2020.11.14 09:38:07.937274 [ 2343 ] {} <Trace> Application: Initialized DateLUT with time zone 'Asia/Tokyo'.
2020.11.14 09:38:07.937310 [ 2343 ] {} <Debug> Application: Setting up /usr/local/ymir/clickhousevar/data/tmp/ to store temporary data in it
2020.11.14 09:38:07.938293 [ 2343 ] {} <Debug> Application: Configuration parameter 'interserver_http_host' doesn't exist or exists and empty. Will use '9913-067' as replica host.
2020.11.14 09:38:07.941001 [ 2343 ] {} <Debug> ConfigReloader: Loading config '/usr/local/ymir/clickhousevar/config/users.xml'
```

**Comment by amosbird** [comment]
It's possibly fixed by  https://github.com/ClickHouse-Extras/poco/pull/28
